src/algorithm/model.hxx

```diff
@@ -0,0 +1,211 @@
+          const GeometryModel&,
+          JointIndex,
+          SE3,
+          ModelTpl<Scalar,Options,JointCollectionTpl> &,
+          GeometryModel& > ArgsType;
+
+      template <typename JointModel>
+      static void algo (const JointModelBase<JointModel> & jmodel,
+          const ModelTpl<Scalar,Options,JointCollectionTpl> & modelAB,
+          const GeometryModel& geomModelAB,
+          JointIndex parentId,
+          SE3 pMi,
+          ModelTpl<Scalar,Options,JointCollectionTpl> & model,
+          GeometryModel& geomModel)
+      {
+        // If old parent is universe, use what's provided in the input.
+        // otherwise, get the parent from modelAB.
+        if (modelAB.parents[jmodel.id()] > 0)
+          parentId = model.getJointId(modelAB.names[modelAB.parents[jmodel.id()]]);
+        JointIndex jid = model.addJoint (
+            parentId,
+            jmodel,
+            pMi * modelAB.jointPlacements[jmodel.id()],
+            modelAB.names[jmodel.id()],
+            jmodel.jointVelocitySelector(modelAB.effortLimit),
+            jmodel.jointVelocitySelector(modelAB.velocityLimit),
+            jmodel.jointConfigSelector(modelAB.lowerPositionLimit),
+            jmodel.jointConfigSelector(modelAB.upperPositionLimit));
+        assert (jid < model.joints.size());
+
+        model.appendBodyToJoint (jid, modelAB.inertias[jmodel.id()]);
+        // Add all frames whose parent is this joint.
+        for (FrameIndex fid = 1; fid < modelAB.frames.size(); ++fid) {
+          Frame frame = modelAB.frames[fid];
+          if (frame.parent == jmodel.id()) {
+            frame.parent = jid;
+            assert (frame.previousFrame > 0 || frame.type == JOINT);
+            if (frame.previousFrame != 0) {
+              frame.previousFrame = model.getFrameId (
+                  modelAB.frames[frame.previousFrame].name,
+                  modelAB.frames[frame.previousFrame].type);
+            }
+            model.addFrame (frame);
+          }
+        }
+        // Add all geometries whose parent is this joint.
+        for (GeomIndex gid = 0; gid < geomModelAB.geometryObjects.size(); ++gid) {
+          GeometryObject go = geomModelAB.geometryObjects[gid];
+          if (go.parentJoint == jmodel.id()) {
+            go.parentJoint = jid;
+            assert (go.parentFrame > 0);
+            if (go.parentFrame != 0) {
+              go.parentFrame = model.getFrameId (
+                  modelAB.frames[go.parentFrame].name,
+                  modelAB.frames[go.parentFrame].type);
+            }
+            geomModel.addGeometryObject (go);
+          }
+        }
+      }
+    };
+  }
+
+  template<typename Scalar, int Options, template<typename,int> class JointCollectionTpl>
+  void
+  appendModel(const ModelTpl<Scalar,Options,JointCollectionTpl> & modelA,
+              const ModelTpl<Scalar,Options,JointCollectionTpl> & modelB,
+              const GeometryModel& geomModelA,
+              const GeometryModel& geomModelB,
```

Comment:
This algorithm appends models and geometry models at the same time. Not everybody who is using models also uses geometry models. Can you separate these?

Reply:
I am not gonna split this into two algorithms. It's more work and I never received any request in that direction.

If you do not have geometry models, you can pass empty models. If requested later, we can add a prototype without the two GeometryModel.

Reply:
What about this one ?

Reply:
Yes, I agree with @gabrielebndn. It would be worthy to split it. But I do not know if you are willing to provide this additional effort?

Reply:
I was about to propose to add a folder like `unsupported` or `wip` for stuff that has not yet reached the desired level of maturity.

I added an API without the geometry models.

Reply:
OK. But maybe it should be put in the corpus of the doc, and maybe also as a warning during the compilation of the function, for instance, to inform the user about the status of the functionality.

Reply:
In the documentation, it is easy to put it in a separate doc, like Eigen does.
I don't think a compilation warning is a good idea. Adding `#include <pinocchio/wip/algorithm/model.hpp>`, eventually in the namespace `pinocchio::wip::...` is explicit enough.

Reply:
Another simpler option is to add a doxygen macro (called alias in Doxygen syntax) like "this_feature_is_not_mature_yet" which would either:
- put the function into a special page,
- or add a clear warning message.

Reply:
If you ever do this, please do not call it wip, I can already see lots of confusion with robotpkg/wip ("it's in the wip" ^^). "unsupported" is clear enough and Eigen-like